internal/storage/pmem/pmem.go

```diff
@@ -44,10 +48,84 @@ func Mount(ctx context.Context, device uint32, target string) (err error) {
 			osRemoveAll(target)
 		}
 	}()
-	source := fmt.Sprintf("/dev/pmem%d", device)
+	source := fmt.Sprintf(pMemFmt, device)
 	flags := uintptr(unix.MS_RDONLY)
 	if err := unixMount(source, target, "ext4", flags, "noload"); err != nil {
 		return errors.Wrapf(err, "failed to mount pmem device %s onto %s", source, target)
 	}
 	return nil
 }
+
+// MountDM creates dm-linear block device and mounts it on `target`
+func MountDM(ctx context.Context, device uint32, deviceStart, deviceSize int64, target string) (err error) {
+	_, span := trace.StartSpan(ctx, "pmem::MountDM")
+	defer span.End()
+	defer func() { oc.SetSpanStatus(span, err) }()
+
+	span.AddAttributes(
+		trace.Int64Attribute("device", int64(device)),
+		trace.Int64Attribute("deviceStart", deviceStart),
+		trace.Int64Attribute("sectorSize", deviceSize),
+		trace.StringAttribute("target", target))
+
+	if err := osMkdirAll(target, 0700); err != nil {
+		return err
+	}
+	defer func() {
+		if err != nil {
+			osRemoveAll(target)
+		}
+	}()
+
+	lt := LinearTargetPMem(device, deviceSize, deviceStart)
+	linearDeviceName := fmt.Sprintf(linearDeviceFmt, device, deviceStart, deviceSize)
+
+	var dmpath string
+	dmpath, err = dm.CreateDevice(linearDeviceName, dm.CreateReadOnly, []dm.Target{lt})
+	if err != nil {
+		return errors.Wrapf(err, "failed to create dm-linear target: pmem device: %d, offset: %d", device, deviceStart)
+	}
+
+	defer func() {
+		if err != nil {
+			dm.RemoveDevice(linearDeviceName)
+		}
+	}()
```

Comment:
I think you could just do all of this and then call `Mount` above, right?

Reply:
yeah, good point. we could have something like `mountInternal(ctx context.Context, source, target string) {...}` and have `Mount` do `mountInternal(ctx, fmt.Sprintf(pmemfmt, device), target)`